kernel/include/ksched.h

```diff
@@ -71,9 +70,12 @@ void z_ready_thread(struct k_thread *thread);
 void z_requeue_current(struct k_thread *curr);
 struct k_thread *z_swap_next_thread(void);
 void z_thread_abort(struct k_thread *thread);
-void move_thread_to_end_of_prio_q(struct k_thread *thread);
+void z_yield(struct k_thread *curr);
 bool thread_is_sliceable(struct k_thread *thread);
 
+/* This routine is for testing purposes only */
+void z_yield_testing_only(struct k_thread *curr);
+
```

Comment:
Technically no need to declare this in a header if the prototype is declared in tests.